packages/components/src/date-time/date-time/test/index.tsx

```diff
@@ -0,0 +1,198 @@
+/**
+ * External dependencies
+ */
+import { render, screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import timezoneMock from 'timezone-mock';
+
+/**
+ * WordPress dependencies
+ */
+import { getSettings, setSettings, type DateSettings } from '@wordpress/date';
+
+/**
+ * Internal dependencies
+ */
+import DateTimePicker from '..';
+
+describe( 'DateTimePicker', () => {
+	let originalSettings: DateSettings;
+	beforeAll( () => {
+		originalSettings = getSettings();
+		setSettings( {
+			...originalSettings,
+			timezone: {
+				offset: -5,
+				offsetFormatted: '-5',
+				string: 'America/New_York',
+				abbr: 'EST',
+			},
+		} );
+	} );
+
+	afterEach( () => {
+		jest.restoreAllMocks();
+		timezoneMock.unregister();
+	} );
+
+	afterAll( () => {
+		setSettings( originalSettings );
+	} );
+
+	it( 'should display and select dates correctly when timezones match', async () => {
+		const user = userEvent.setup();
+		const onChange = jest.fn();
+
+		timezoneMock.register( 'US/Eastern' );
+
+		const { rerender } = render(
+			<DateTimePicker
+				currentDate="2025-11-15T00:00:00"
+				onChange={ onChange }
+			/>
+		);
+
+		expect(
+			screen.getByRole( 'button', {
+				name: 'November 15, 2025. Selected',
+			} )
+		).toBeInTheDocument();
```

Comment:
Nit, especially for this test, but when testing visible items, I always prefer `toBeVisible()`, since `.toBeInTheDocument()` could be invisible to the user.

```suggestion
		).toBeVisible();
```

For negative queries, `.not.toBeInTheDocument()` is fine, since `isVisible()` will error.

Reply:
I had your voice in my head as I was including these, and came to the same conclusion about the negated assertions (we don't want them in the document, whether or not they're visible), but I guess I neglected to pay as much mind to the positive assertions 😅 I'll update.

Reply:
Updated positive assertions to use `toBeVisible` in ac743c6e9eb.